datacube/drivers/postgres/_api.py

```diff
@@ -521,7 +527,7 @@ def search_datasets_query(expressions, source_exprs=None,
 
         if select_fields:
             select_columns = tuple(
-                f.alchemy_expression.label(f.name)
+                f.alchemy_expression.label(f.name) if f is not None else None
```

Comment:
Is this really necessary considering that we're already inside `if select_fields`? Even if for whatever reason there's a None value in `select_fields`, I feel it would be better to filter it out rather than have a null `select_columns` value that we then need to keep accounting for.

Reply:
Consider a user requesting all fields and not filtering by product.  Some products will have fields that other products do not.  We construct a separate query for each potentially matching product.  Passing down `None`s in the select list down the stack when querying products that we know do not have that field means that we get matching tuple fields for each product - the alternative is even more complex result re-writing at the top of the stack.